features/onboarding/screens/onboarding-contact-card-screen.tsx

```diff
@@ -42,15 +103,25 @@ export function OnboardingContactCardScreen() {
         throw new Error("No Privy user found, please logout");
       }
 
-      await createUserAsync({
-        inboxId: currentSender?.inboxId,
-        privyUserId: privyUser?.id,
-        smartContractWalletAddress: currentSender?.ethereumAddress,
+      // Create and validate the request payload
+      const payload = {
+        inboxId: currentSender.inboxId,
+        privyUserId: privyUser.id,
+        smartContractWalletAddress: currentSender.ethereumAddress,
         profile: {
-          name: useOnboardingContactCardStore.getState().name,
+          name: store.name,
+          username: store.username,
         },
-      });
+      };
 
+      // Validate the payload against our schema
+      const validationResult = createUserRequestSchema.safeParse(payload);
+
+      if (!validationResult.success) {
+        throw new Error("Invalid request data. Please check your input.");
+      }
```

Comment:
looks like there's a loooot of validation, could we have just this one ? and it will handle the profile validation that we do just above?